internal/crypto/cert.go

```diff
@@ -59,3 +67,94 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// MergeCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func MergeCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	// get cert bundle path
+	if pluginConfig == nil {
+		return originalCerts, nil
+	}
+	certBundlePath, ok := pluginConfig[certBundleKey]
+	if !ok {
+		return originalCerts, nil
+	}
+
+	// validate certificate bundle path
+	basename := filepath.Base(certBundlePath)
+	if !isValidFileName(basename) {
+		return nil, fmt.Errorf("the filename of certificate bundle path is not cross-platform compatible. filename: %s", basename)
+	}
+
+	// read certificate bundle
+	certBundleBytes, err := os.ReadFile(certBundlePath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open user provided cert bundle file %q with err: %w", certBundlePath, err)
+	}
+	certBundle, err := parsePEM(certBundleBytes)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse user provided cert bundle file %q with err: %w", certBundlePath, err)
+	}
+	// merge certificate chain
+	return mergeCertChain(originalCerts, certBundle)
+}
+
+// mergeCertChain appends the parent certificate from the certificate bundle to the original certificate chain.
+// It returns the complete certificate chain along with a nil error if successful, otherwise it returns a nil slice and a non-nil error indicating the failure reason.
+func mergeCertChain(originalCerts, certBundle []*x509.Certificate) ([]*x509.Certificate, error) {
+	if len(originalCerts) == 0 {
+		return nil, errors.New("the certificate chain accessed from Azure Key Vault is empty")
+	}
+	if len(certBundle) == 0 {
+		return nil, errors.New("the certificate bundle file is empty or in a unsupported format")
+	}
+	// leaf certificate
+	leafCert := originalCerts[0]
```

Comment:
nit: 
we could define variable right before we use it

Reply:
Updated.